### PR TITLE
Prevent bors from merging into any branch except master

### PR DESCRIPTION
**Pull request by rvl**
Another try at #1632.

### CODE REVIEW

**Review by KtorZ — APPROVED** [review]

**Comment by KtorZ** [comment]
:point_up: should fail.

**Comment by KtorZ** [comment]
--> https://buildkite.com/input-output-hk/cardano-wallet/builds/7370#edb2b04f-54c1-4156-812f-9cac2b67ac3d

So this fails, but the rest of the job is still running. Is there a "fail fast" option on bors :thinking: ? Or on buildkite actually, because bors has probably no power here and is simply waiting for Buildkite to notify.

**Comment by KtorZ** [comment]
Seems like we could mark all other steps to `depends_on` this one, to make sure they only run if necessary.

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -5,11 +5,13 @@ env:
...
-  - label: 'hydra-eval-errors'
-    command: 'nix-build ./nix -A iohkNix.hydraEvalErrors && ./result/bin/hydra-eval-errors.py'
-    if: 'build.tag == null'
+  - label: 'Prevent merging to wrong branch'
+    if: 'build.branch == "bors/staging"'
+    command: './.buildkite/check-bors.sh'
     agents:
       system: x86_64-linux
+  - wait: ~
+    if: 'build.branch == "bors/staging"'
```
Comment (KtorZ) [inline .buildkite/pipeline.yml:14]: Also :+1:

**Review by KtorZ — APPROVED** [review]

**Comment by rvl** [comment]
I added a conditional wait step which should make it fail fast.
May need to wait a little while to test because buildkite are doing some unscheduled db maintenance.

**Comment by KtorZ** [comment]
Hydra timing out again.

**Pull request merged**